Sources/Controllers/2.0/OAPOIFilterViewController.mm

```diff
@@ -360,10 +360,31 @@ - (void)setupPopover:(UIAlertController *)alert {
     popPresenter.sourceRect = _btnMore.frame;
     popPresenter.permittedArrowDirections = UIPopoverArrowDirectionUp;
 }
-- (void) editCategories
+
+- (void)showEditCategoriesScreen
+{
+    OACustomPOIViewController *customPOIScreen = [[OACustomPOIViewController alloc] initWithFilter:_filter];
+    customPOIScreen.delegate = self;
+    customPOIScreen.modalPresentationStyle = UIModalPresentationFullScreen;
+    [self showViewController:customPOIScreen];
+}
+
+#pragma mark - OACustomPOIViewDelegate
+
+- (void)searchByUIFilter:(OAPOIUIFilter *)filter wasSaved:(BOOL)wasSaved
+{
+    [self.navigationController popViewControllerAnimated:YES];
+    [self.customPOIDelegate searchByUIFilter:filter wasSaved:wasSaved];
```

Comment:
Check delegate for nil here and in other places, if the delegate is a weak reference, we should always chack for nil